src/translate.py

```diff
@@ -1942,9 +1940,23 @@ def __str__(self) -> str:
 
 @dataclass
 class RegMeta:
+    """Metadata associated with a register value assignment.
+
+    This could largely be computed ahead of time based on instr_inputs/
+    instr_uses, except for the fact that function pointer argument passing
+    is determined dynamically based on type info.
+
+    Except for a few properties, metadata is propagated across blocks only
+    after translation is done (because parent block metadata is not generally
+    available when constructing reg metas)."""
```

Comment:
Maybe mention `propagate_register_meta` here?
This is sort of related to something I brought up in the Discord recently, that "translation" is a bit ambiguous and it'd be nice to have terms for the sub-phases. (The propagation here happens after block translation, but during `translate_to_ast`...)

Reply:
Done, went with the term "block translation" for now.